ui/de-lib/src/main/java/org/iplantc/de/apps/client/views/AppsViewImpl.java

```diff
@@ -25,39 +30,42 @@
     }
 
     @UiField(provided = true) final AppsToolbarView toolBar;
-    @UiField(provided = true) final AppCategoriesView appCategoriesView;
+    @UiField TabPanel categoryTabs;
+    AppCategoriesView.Presenter categoriesPresenter;
+    OntologyHierarchiesView.Presenter hierarchiesPresenter;
+
     @UiField(provided = true) final AppsGridView appsGridView;
 
     private static final MyUiBinder uiBinder = GWT.create(MyUiBinder.class);
 
     @Inject
     protected AppsViewImpl(@Assisted final AppCategoriesView.Presenter categoriesPresenter,
+                           @Assisted final OntologyHierarchiesView.Presenter hierarchiesPresenter,
                            @Assisted final AppsGridView.Presenter gridPresenter,
                            @Assisted final AppsToolbarView.Presenter toolbarPresenter) {
-        this.appCategoriesView = categoriesPresenter.getView();
+        this.categoriesPresenter = categoriesPresenter;
+        this.hierarchiesPresenter = hierarchiesPresenter;
         this.appsGridView = gridPresenter.getView();
         this.toolBar = toolbarPresenter.getView();
 
         initWidget(uiBinder.createAndBindUi(this));
+        categoryTabs.addSelectionHandler(new SelectionHandler<Widget>() {
+            @Override
+            public void onSelection(SelectionEvent<Widget> event) {
+                Widget selectedItem = event.getSelectedItem();
+                for (Widget currentItem : categoryTabs) {
+                    if (currentItem != selectedItem) {
+                        ((Tree)currentItem).getSelectionModel().deselectAll();
```

Comment:
Doing something similar in the Hierarchies Presenter I think... probably don't need this one as well.